aion/lib/aion/ranking.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Aion.Ranking do
+  @moduledoc """
+  This module provides logic for generating rankings
+  """
+  import Ecto.Query, only: [from: 2]
+  alias Aion.{Repo, User, UserCategoryScore}
+
+  @type user_general_score :: %{user_name: String.t, score: integer}
+
+  @spec general :: list user_general_score
```

Comment:
Also, is there a reason why you didn't use Ecto.Schema? You wouldn't have to do most of this stuff manually I guess (same for the migration generation etc.)

Reply:
Hmm. I am not sure what you mean.
Basically, we are dynamically fetching data which is the result of some kind of computation on specific tables.
If you could explain it, i'd be glad ;)

Reply:
As @pmrukot mentioned, this is how we are dealing with _all_ of the models we have in our repo. Most of the stuff you implemented would be generated using `Phoenix`'s scaffolding tools. You can read more about it [here](https://hexdocs.pm/phoenix/1.1.0/Mix.Tasks.Phoenix.Gen.Model.html) as one of the generated files, it provides the actual `Ecto.Schema` :)

Reply:
Hmm, but this is note the physical model. It is just the presentation of calculated data.
We don't have ranking table in our system but we have it for players for example.

Reply:
That's a fair point, I totally forgot about it. Then it shouldn't be `Ecto.Schema` I guess.